src/devices/CharacterLcd/LcdRgb1602.cs

```diff
@@ -0,0 +1,154 @@
+        /// <param name="rgbDevice">The I2C device to control RGB backlight.</param>
+        public LcdRgb1602(I2cDevice lcdDevice, I2cDevice rgbDevice)
+            : base(new Size(16, 2))
+        {
+            _lcdDevice = lcdDevice;
+            _rgbDevice = rgbDevice;
+
+            Initialize(Size.Height);
+            InitRgb();
+        }
+
+        public void Dispose()
+        {
+            if (_lcdDevice != null)
+            {
+                _lcdDevice.Dispose();
+                _lcdDevice = null;
+            }
+
+            if (_rgbDevice != null)
+            {
+                _rgbDevice.Dispose();
+                _rgbDevice = null;
+            }
+        }
+
+        /// <summary>
+        /// Enable/disable the backlight.
+        /// </summary>
+        public override bool BacklightOn {
+            get { return _backlightOn; }
+
+            set
+            {
+                _backlightOn = value;
+
+                if (_backlightOn)
+                {
+                    ForceSetColor(_currentColor);
+                }
+                else
+                {
+                    ForceSetColor(Color.Black);
+                }
+            }
+        }
+
+        /// <summary>
+        /// Sets register for RGB backlight.
+        /// </summary>
+        /// <param name="addr">The register address.</param>
+        /// <param name="value">The register value.</param>
+        private void SetRgbRegister(RgbRegisters addr, byte value)
+        {
+            _rgbDevice.Write(new byte[] { (byte)addr, value });
+        }
+
+        /// <summary>
+        /// Sets the backlight color without any checks.
+        /// </summary>
+        /// <param name="color">The color to set.</param>
+        private void ForceSetColor(Color color)
+        {
+            SetRgbRegister(RgbRegisters.REG_RED, color.R);
+            SetRgbRegister(RgbRegisters.REG_GREEN, color.G);
+            SetRgbRegister(RgbRegisters.REG_BLUE, color.B);
+        }
+
+        /// <summary>
+        /// Initializes RGB device.
+        /// </summary>
+        private void InitRgb()
+        {
+            // backlight init
+            SetRgbRegister(RgbRegisters.REG_MODE1, 0);
+
+            // set LEDs controllable by both PWM and GRPPWM registers
+            SetRgbRegister(RgbRegisters.REG_LEDOUT, 0xFF);
+
+            // set MODE2 values
+            // 0010 0000 -> 0x20  (DMBLNK to 1, ie blinky mode)
+            SetRgbRegister(RgbRegisters.REG_MODE2, 0x20);
+
+            SetColor(Color.White);
+        }
+
+        /// <summary>
+        /// Send a data or command byte to the controller.
+        /// </summary>
+        /// <param name="data">True to send data, otherwise sends a command.</param>
+        protected override void Send(byte value, bool data = false)
+        {
+            _lcdDevice.Write(new byte[] { data ? (byte)0x40 : (byte)0x80, (byte)value });
```

Comment:
you can avoid allocation on each send by doing:

```csharp
ReadOnlySpan<byte> dataToSend = stackalloc byte[2] { data ? ... };
```

Reply:
Fixed.